src/nvim/regexp.c

```diff
@@ -3293,7 +3293,8 @@ bt_regexec_nl (
   ireg_ic = rmp->rm_ic;
   ireg_icombine = FALSE;
   ireg_maxcol = 0;
-  return bt_regexec_both(line, col, NULL) != 0;
+
+  return bt_regexec_both(line, col, NULL);
```

Comment:
`bt_regexec_both`'s return value is determined by `regtry`, which returns either `0 for failure, number of lines contained in the match otherwise.` Given that, I think an assert followed by a cast is reasonable.